RunningOrder/Shared/Components/Search/SearchBarSuggestions+Logic.swift

```diff
@@ -11,37 +11,94 @@ import Combine
 import SwiftUI
 
 extension SearchBarSuggestions {
+    enum Scope: CaseIterable {
+        case allSprints
+        case currentSprint
+
+        var title: LocalizedStringKey {
+            switch self {
+            case .allSprints:
+                return "All Sprints"
+            case .currentSprint:
+                return "Current Sprint"
+            }
+        }
+    }
     final class Logic: ObservableObject {
         private unowned var storyManager: StoryManager
+        private unowned var sprintManager: SprintManager
+        private unowned var appStateManager: AppStateManager
+        private unowned var userManager: UserManager
+
         let inputText: String
 
-        init(inputText: String, storyManager: StoryManager) {
+        @Published var searchScope: Scope = .allSprints
+
+        init(storyManager: StoryManager, sprintManager: SprintManager, appStateManager: AppStateManager, userManager: UserManager, inputText: String) {
             self.storyManager = storyManager
+            self.sprintManager = sprintManager
+            self.appStateManager = appStateManager
+            self.userManager = userManager
             self.inputText = inputText
         }
 
-        var filteredStories: [Story] {
-            return storyManager.allStories
+        var filteredSearchSections: [SearchSection] {
+            guard case .spaceSelected(let space) = appStateManager.currentState else {
+                return []
+            }
+
+            let allSprints = sprintManager.sprints(for: space.id)
+            let selectedSprints: [Sprint.ID]
+            switch searchScope {
+            case .allSprints:
+                selectedSprints = allSprints.map { $0.id }
+            case .currentSprint:
+                if let first = allSprints.first { // TODO: change this sprint to the selected Sprint
+                    selectedSprints = [first.id]
+                } else {
+                    return []
```

Comment:
do you really want to return an empty array?

Reply:
well having `allSprints.first` failed implies that the allSprints array are empty, so in all cases, it will be empty, so yes